admin/src/js/controllers/multiple-user.js

```diff
@@ -0,0 +1,151 @@
+angular.module('controllers').controller('MultipleUserCtrl', function(
+  $scope,
+  $uibModalInstance,
+  $window,
+  CreateMultipleUser,
+  DB
+) {
+
+  'use strict';
+  'ngInject';
+
+  // $scope.hideFooter = true;
+  $scope.status = { uploading: false };
+  $scope.displayAddMultipleModal = true;
+  $scope.displayUnavailableModal = false;
+  $scope.displayUploadConfirm = false;
+  $scope.displayProcessingStatus = false;
+  $scope.displayFinishSummary = false;
+  $scope.outputFileUrl = '';
+  $scope.processTotal = '';
+  const USER_LOG_DOC_ID = 'bulk-user-upload-';
+
+  $scope.onCancel = function () {
+    $scope.clearScreen();
+    $uibModalInstance.dismiss();
+  };
+
+  const getLogsByType = (docPrefix) => {
+    return DB({ logsDB: true })
+      .allDocs({
+        startkey: docPrefix,
+        endkey: docPrefix + '\ufff0',
+        include_docs: true
+      })
+      .then(result => {
+        if (!result || !result.rows || !result.rows.length) {
+          return;
+        }
+        return result.rows
+          .map(row => row.doc)
+          .sort((a, b) => new Date(b.bulk_uploaded_on) - new Date(a.bulk_uploaded_on));
+      });
+  };
+
+  const prepareStringForCSV = (str) => {
+    if (!str) {
+      return str;
+    }
+    return '"' + str.replace(/"/g, '""') + '"';
+  };
+
+  const convertToCSV = doc => {
+    if (!doc || !doc.data) {
+      return;
+    }
+    const eol = '\r\n';
+    const delimiter = ',';
+    const columns = ['import.status:excluded', 'import.message:excluded', 'import.username:excluded'];
+    let output = columns.join(delimiter) + eol;
+
+    doc.data.forEach(record => {
+      if (record.import) {
+        output += [
+          prepareStringForCSV(record.import.status),
+          prepareStringForCSV(record.import.message),
+          prepareStringForCSV(record.username),
+        ].join(delimiter) + eol;
+      }
+    });
+
+    const file = new Blob([output], { type: 'text/csv;charset=utf-8;' });
```

Comment:
It is preferable to stream data in cases like this to avoid building it all up in memory. If it's data from the client though, the only option I know of is [Blob.stream](https://developer.mozilla.org/en-US/docs/Web/API/Blob/stream), but that needs newer browser support. If we can do that, it is preferred. If we cannot, the current in-memory implementation may still be fine.